src/opamWixMain.ml

```diff
@@ -109,6 +109,70 @@ module Args = struct
 
 end
 
+module type FILE_INTF = sig
+  type t
+  val name : string
+  val to_string : t -> string
+  val of_string : string -> t
+  val (/) : dirname -> string -> t
+  val copy : src:t -> dst:t -> unit
+  val exists : t -> bool
+  val basename : t -> basename
+end
+
+module Dir_impl : FILE_INTF
+  with type t = OpamFilename.Dir.t = struct
+  include OpamFilename.Dir
+  let name = "directory"
+  let (/) = OpamFilename.Op.(/)
+  let copy = OpamFilename.copy_dir
+  let exists = OpamFilename.exists_dir
+  let basename = OpamFilename.basename_dir
+end
+
+module File_impl : FILE_INTF
+  with type t = OpamFilename.t = struct
+  include OpamFilename
+  let name = "file"
+  let (/) = OpamFilename.Op.(//)
+  let copy = OpamFilename.copy
+  let exists = OpamFilename.exists
+  let basename = OpamFilename.basename
+end
+
+let resolve_path (type a) env
+  (module F : FILE_INTF with type t = a) path =
+  let expnaded_path = OpamFilter.expand_string env path in
```

Comment:
```suggestion
  let expanded_path = OpamFilter.expand_string env path in
```

Reply:
I didn't understand what you wanted to say